src/match.c

```diff
@@ -1,128 +1,199 @@
 #include <stdio.h>
 #include <stdlib.h>
-#include <stdbool.h>
-#include <assert.h>
-#include <gmodule.h>
+#include <string.h>
+#include "simclist.c"
```

Comment:
Never `#include` `.c` files. I'm surprised `gcc` isn't yelling at y'all, this should generate a lot of errors.

Reply:
@mpcsh I put this in initially to help compile the match library, since the -c flag doesn't allow for multiple files to be compiled that way. I'm still unsure of how I would compile match.c and simclist.c with -c properly to make the library for testing.